sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -210,7 +211,8 @@ private void drainQueue() {
                 if (!isEmitted) {
                     // The only reason we can't emit was the downstream(s) were terminated hence nobody
                     // to receive the message.
-                    if (isPrefetchDisabled) {
+                    // In RECEIVE_AND_DELETE mode, we re-buffer the message because it is deleted in service side.
+                    if (isPrefetchDisabled && asyncClient.getReceiverOptions().getReceiveMode() == ServiceBusReceiveMode.PEEK_LOCK) {
```

Comment:
Given drain-loop is a hot path, I wonder if we should avoid hopping through objects to check the receive mode. We could have a boolean property in the class level (e.g., `isPeekLockReceiveMode`) and use it here instead.

Reply:
Maybe pass this in as a constructor parameter since it is a read-only option that is set at client creation time

Reply:
Good idea, I'll change this.